### PR TITLE
Add optional Queryparameter to getAllRoles Service

### PR DESCRIPTION
**Pull request by marco2704**
According to improve this method without change its previous behavior, a new Queryparameter was added to getAllRoles method in RoleBasedAuthorizationStrategy. It may be useful in some case.

Thanks!

### CODE REVIEW

File: src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java [hunk 1]
```diff
@@ -424,25 +424,27 @@ public void doUnassignRole(@QueryParameter(required = true) String type,
...
     /**
      * API method to get all groups/users with their role in globalRoles
-     * Example: curl -X localhost:8080/role-strategy/strategy/getAllRoles
+     * Example: curl -X GET localhost:8080/role-strategy/strategy/getAllRoles?type=projectRoles
      *
-     * @throws IOException in case data fails
-     * @since 2.6.0
+     * @param type (globalRoles by default, projectRoles, slaveRoles)
+     *
+     * @since 2.8.0
```
Comment (oleg-nenashev) [inline src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java]: No need to modify that. The signature changed, but from Stapler PoV the methods are compatible

Comment (marco2704) [reply src/main/java/com/michelin/cio/hudson/plugins/rolestrategy/RoleBasedAuthorizationStrategy.java]: Right, changed.

**Comment by marco2704** [comment]
Hi! Another option with less changes:
```java 
/**
     * API method to get all groups/users with their role in globalRoles
     * Example: curl -X localhost:8080/role-strategy/strategy/getAllRoles?type=projectRoles
     *
     * @throws IOException in case data fails
     * @since 2.6.0
     */
    @Restricted(NoExternalUse.class)
    public void doGetAllRoles(StaplerRequest req, StaplerResponse rsp) throws IOException {
        checkAdminPerm();
        req.setCharacterEncoding("UTF-8");
        OutputStream os = rsp.getOutputStream();
        JSONObject json = new JSONObject();
        String type = req.hasParameter("type") ? req.getParameter("type") : GLOBAL;
        RoleMap roleMap = this.grantedRoles.get(type);
        for (Map.Entry<Role, Set<String>> grantedRole : roleMap.getGrantedRoles().entrySet()) {
          json.put(grantedRole.getKey().getName(), grantedRole.getValue());
        }
        os.write(json.toString().getBytes("UTF-8"));
    }
```

Both of them were tested in a staging Jenkins instance. Let me know any advice or improvement.

Thanks in advance!

**Comment by oleg-nenashev** [comment]
I am fine with the current implementation. Generally it would be better to use https://github.com/stapler/stapler/blob/311dcfd92d229569e7aaaaae364f6e702188fddc/core/src/main/java/org/kohsuke/stapler/json/JsonHttpResponse.java so that the content type is set correctly.

@marco2704 Would it be possible to do so?

**Comment by marco2704** [comment]
I'm not sure but i think it's not possible yet, because of the stapler version which seems to be 1.237 where json package does not exist.

**Comment by marco2704** [comment]
Hi @oleg-nenashev! I've added a similar behavior, just until we can use [JsonHttpResponse.java](https://github.com/stapler/stapler/blob/311dcfd92d229569e7aaaaae364f6e702188fddc/core/src/main/java/org/kohsuke/stapler/json/JsonHttpResponse.java) , Regards!

**Comment by marco2704** [comment]
Hi @oleg-nenashev any update about it? Hope we can merge it  😁

**Comment by oleg-nenashev** [comment]
My bad, I missed the notifications. Merging/releasing

**Pull request merged**